code/__HELPERS/medals.dm

```diff
@@ -0,0 +1,119 @@
+//The code here is used for BYOND medals, typically awarded for things like killing megafauna or winning a pulse rifle.
+//It also covers stuff for keeping track of certain statistics.
+
+#define ANNOUNCE_TO_NOBODY "nobody" //Silent, doesn't tell anyone
+#define ANNOUNCE_TO_PLAYER "player" //Tells the person who earned the achievement
+#define ANNOUNCE_TO_ADMINS "admins" //Tells the unlocker and any online admins
+#define ANNOUNCE_TO_EVERYONE "everyone" //Tells everyone
+
+/proc/CheckMedal(medal,client/player) //Checks if a player has a medal, returning 1 on a success and 0 on a failure
+
+	if(!player || !medal)
+		return
+	if(global.medal_hub && global.medal_pass && global.medals_enabled)
+
+		var/result = world.GetMedal(medal, player, global.medal_hub, global.medal_pass)
+
+		if(isnull(result))
+			global.medals_enabled = FALSE
+			log_game("MEDAL ERROR: Could not contact hub to get medal:[medal] player:[player.ckey]")
+			message_admins("Error! Failed to contact hub to get [medal] medal for [player.ckey]!")
+			return 0
+		else if (result)
+			return TRUE
+		return FALSE
+
+
+/proc/UnlockMedal(medal, client/player, announce = ANNOUNCE_TO_PLAYER) //Adds a medal to the player
+	if(!player || !medal)
+		return
+	if(global.medal_hub && global.medal_pass && global.medals_enabled)
+		spawn()
+			var/result = world.SetMedal(medal, player, global.medal_hub, global.medal_pass)
+			if(isnull(result))
+				global.medals_enabled = FALSE
+				log_game("MEDAL ERROR: Could not contact hub to award medal:[medal] player:[player.ckey]")
+				message_admins("Error! Failed to contact hub to award [medal] medal to [player.ckey]!")
+			else
```

Comment:
This is wrong. I had the else if(result) there for a reason.
If the medal is already unlocked, the hub returns 0. With this else, it will announce it every time.